### DISCUSSION TITLE
How to convert string to ViewModel object?

### DISCUSSION THREAD

**Discussion by jsimunac01** [canton7/Stylet#377] [Q&A]
I've got a reporting dashboard.  When the user clicks a button, it opens the correct viewmodel and displays the view.  So button1 open VM1ViewModel, button2 opens VM2ViewModel, etc.  This means that all possible viewmodels are hard coded in the program.  I would like to store the viewmodel names in the database (with the rest of the user interface information).  That is stored as a string.  So for button1 I would pass "VM1ViewModel" as string.  Now I need to open it with WindowManager.ShowWindow(new VM1ViewModel).  If I just type the name, no problem.  But I can't use the string.  I've tried converting to a Screen object, but I can't find any way to convert from a string to Screen.  Is this even possible?  If so, how would I go about it.  Google is being singularly unhelpful on this one.  Thanks in advance.

**Comment by canton7** [comment]
Such use of strings instead of types, or "stringy-typed", is usually discouraged - you lose any compile-time help the compiler can give you, and errors will appear at runtime instead.

That said, you could use a dictionary to map between the string and the type, or use `Type.GetType` or `Assembly.GetType` to get the type for a fully-qualified type name, and `Activator.CreateInstance` to instantiate it.

**Reply by jsimunac01** [reply]
Thanks - I'll play with that and see how it works.  I'm just trying to get rid of huge If or Switch statements to wade through all the possible viewmodels.

**Reply by jsimunac01** [reply]
I should probably ask (to save myself a lot of trial and error), what type should I use for the ViewModel in the dictionary?  Not Screen (I already tried that).
